test/unit/Util/FileHelperTest.php

```diff
@@ -26,4 +26,17 @@ public function testSystemWindowsPath() : void
 
         self::assertSame(strtr($path, '\\/', DIRECTORY_SEPARATOR . DIRECTORY_SEPARATOR), FileHelper::normalizeSystemPath($path));
     }
+
+    public function testSystemWindowsPathWithProtocol() : void
+    {
+        if (DIRECTORY_SEPARATOR !== '\\') {
+            $this->markTestSkipped('Test runs only on Windows');
+        }
+
+        $path = 'phar://C:/Users/ondrej/phpstan.phar/src/TrinaryLogic.php';
+        self::assertSame(
+            'phar://C:\Users\ondrej\phpstan.phar\src\TrinaryLogic.php',
```

Comment:
Interesting that phars still have borked file paths :shrug:

Reply:
I don't get it - `'phar://C:\Users\ondrej\phpstan.phar\src\TrinaryLogic.php'` is what we want here.

Reply:
Yes, but effectively this is no longer a filesystem path.

Anyway, all good with this patch.

Reply:
Thanks :)